### PR TITLE
feat: add types

### PR DESCRIPTION
**Pull request by zzcan**
完善bundle config类型

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -132,9 +130,8 @@ export interface IFatherBundleConfig extends IFatherBaseConfig {
 
   /**
    * modify webpack config via webpack-chain
-   * @todo  real type definition
    */
-  chainWebpack?: (args: any) => any;
+  chainWebpack?: (memo: any, args: any) => void;
```
Comment (PeachScript) [inline src/types.ts:134]: memo：`import type IWebpackChain from '@umijs/bundler-webpack/compiled/webpack-5-chain';`
args: `{ env: string, webpack: 从 @umijs/bundler-webpack 里引入 }`

File: src/types.ts [hunk 2]
```diff
@@ -53,27 +55,23 @@ export interface IFatherBaseConfig {
 
   /**
    * configure postcss
-   * @todo  real type definition
    */
-  postcssOptions?: any;
+  postcssOptions?: Record<string, any>;
```
Comment (PeachScript) [inline src/types.ts:59]: 尽量从 `@umijs/bundler-webpack` 拿已有类型，后续会在 Umi 层再做一次补全
```ts
import type { IConfig as IBundlerWebpackConfig }  from '@umijs/bundler-webpack/dist/types';

IBundlerWebpackConfig['extraBabelPresets'];
IBundlerWebpackConfig['extraBabelPlugins'];
IBundlerWebpackConfig['postcssLoader'];
```

Comment (zzcan) [reply src/types.ts:59]: 👌🏻

File: src/types.ts [hunk 3]
```diff
@@ -132,9 +132,11 @@ export interface IFatherBundleConfig extends IFatherBaseConfig {
 
   /**
    * modify webpack config via webpack-chain
-   * @todo  real type definition
    */
-  chainWebpack?: (args: any) => any;
+  chainWebpack?: (
+    memo: IWebpackChain,
+    args: { env: string; webpack: Compiler['webpack'] },
+  ) => void;
```
Comment (PeachScript) [inline src/types.ts:139]:
```suggestion
  ) => IWebpackChain;
```

**Pull request merged**